lib/internal/fs/promises.js

```diff
@@ -446,12 +447,33 @@ async function copyFile(src, dest, mode) {
 // Note that unlike fs.open() which uses numeric file descriptors,
 // fsPromises.open() uses the fs.FileHandle class.
 async function open(path, flags, mode) {
+  let options;
+  let parents;
+  // Handle case where 2nd argument is options-bag
+  if (typeof flags === 'object' && flags) {
```

Comment:
📓 similar to https://github.com/nodejs/node/pull/35775/files#r1380873544
```suggestion
  if (typeof flags === 'object') {
```